src/prebid.js

```diff
@@ -1018,16 +1022,36 @@ function processQueue(queue) {
   });
 }
 
+/**
+ * Returns a wrapper around fn that delays execution until the page if activated, if it was prerendered.
+ * https://developer.chrome.com/docs/web-platform/prerender-pages
+ */
+function delayIfPrerendering(fn) {
+  return function () {
+    if (document.prerendering && !(config.getConfig('allowPrerendering') || getGlobal().allowPrerendering)) {
+      const that = this;
+      const args = Array.from(arguments);
+      return new Promise((resolve) => {
+        document.addEventListener('prerenderingchange', () => {
+          logInfo(`Auctions are suspended while page is prerendering. Set $$PREBID_GLOBAL$$.allowPrerendering = true to allow auctions during prerendering.`)
+          resolve(fn.apply(that, args))
+        }, {once: true})
+      })
+    } else {
+      return Promise.resolve(fn.apply(this, arguments));
+    }
+  }
+}
 /**
  * @alias module:pbjs.processQueue
  */
-pbjsInstance.processQueue = function () {
+pbjsInstance.processQueue = delayIfPrerendering(function () {
   pbjsInstance.que.push = pbjsInstance.cmd.push = quePush;
   insertLocatorFrame();
   hook.ready();
   processQueue(pbjsInstance.que);
   processQueue(pbjsInstance.cmd);
-};
+});
```

Comment:
Hi I have a question, since now the `processQueue` is called when the prebid instance is ready, introducing this change will introduce the delay of all the queues, right?

I understood that the delay was actually only related to the `requestBids`, and this seems fantastic because we can execute everything in the **prerendering** "phase" and then firing only the auctions when the user actually views the page. 
Can you tell me if I'm wrong? Sorry in advance.

Reply:
That's a good suggestion. I defaulted to delaying everything because it seemed safer, but we couldn't think of an example of something that would break if we delayed only auctions. Updated.